### DISCUSSION TITLE
Running MicroPython on one core of the ESP32-S3 and native c on another core.

### DISCUSSION THREAD

**Discussion by glmassey** [micropython/micropython#9292] [Hardware & Peripherals] [answered]
The ESP32-S3 SoC is a dual core processor.  I am wondering if it is possible to run MicroPython on one core, while running native C code on the second processor.  I would like to boot into the MicroPython environment, and then launch a thread on the other core.  I am not sure exactly how I would compile a binary that can be launched by MicroPython, would doing so be difficult using the SDK from Espressif?

Also, what mechanisms would be possible for inter-thread communication between the two cores as far as MicroPython is concerned?  Does it have support for normal inter-process communication methods such as semaphores and mailboxes?

Thank you.

**Comment by karfas** [comment, marked as answer]
The thread on the other core needs to be compiled into the micropython binary (see https://github.com/micropython/micropython/tree/master/ports/esp32#readme) and https://forum.micropython.org/viewtopic.php?t=7776 
As far as I know, at the moment there don't exist generic micropython bindings for the various FreeRTOS IPC mechanisms.
